core/base/src/main/java/alluxio/util/io/PathUtils.java

```diff
@@ -121,6 +126,35 @@ public static String concatPath(Object base, Object... paths) throws IllegalArgu
     return output.toString();
   }
 
+  /**
+   * @param paths the set of paths
+   * @return the lowest common ancestor, or null if paths is null or empty
+   */
+  public static AlluxioURI findLowestCommonAncestor(Collection<AlluxioURI> paths) {
+    if (paths == null || paths.isEmpty()) {
+      return null;
+    }
+    List<String> matchedComponents = null;
+    int matchedLen = 0;
+    for (AlluxioURI path : paths) {
+      String[] pathComp = path.getPath().split(AlluxioURI.SEPARATOR);
+      if (matchedComponents == null) {
+        matchedComponents = new ArrayList<>(Arrays.asList(pathComp));
+        matchedLen = pathComp.length;
+      }
+
+      for (int i = 0; i < pathComp.length && i < matchedLen; ++i) {
+        if (!matchedComponents.get(i).equals(pathComp[i])) {
+          matchedLen = i;
+          break;
+        }
+      }
+    }
+    AlluxioURI lca = new AlluxioURI(AlluxioURI.SEPARATOR);
+    return new AlluxioURI(
+        PathUtils.concatPath(lca, matchedComponents.subList(0, matchedLen).toArray()));
```

Comment:
does `matchedComponents` not have an entry for the root?

Reply:
nope, "/".split -> returns an empty list